src/ex4/dist/clients/item_client.js

```diff
@@ -1 +1,78 @@
-// Create an ItemClient class here. This is what makes requests to your express server (your own custom API!)
\ No newline at end of file
+// Create an ItemClient class here. This is what makes requests to your express server (your own custom API!)
+
+const getAllTasks = async () => {
+	try {
+		const res = await fetch("/tasks");
+		if (res.ok) {
+			return res.json();
+		}
+	} catch (err) {
+		return err;
+	}
+};
+
+const addItem = async (item) => {
+	const res = await fetch("/tasks/add", {
+		method: "POST",
+		headers: { "Content-Type": "application/json" },
+		body: JSON.stringify(item),
+	});
+	if (res.ok) {
+		return res.status;
+	} else {
+		return res.status;
+	}
+};
+
+const updateItem = async (itemToUpdate, itemId) => {
+	try {
+		const res = await fetch(`/tasks/update/${itemId}`, {
+			method: "PUT",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify(itemToUpdate),
+		});
+		if (res.ok) {
+			return;
+		} else {
+			throw new Error("not updated");
+		}
+	} catch (err) {
+		return err;
+	}
+};
+
+const deleteItem = async (itemId) => {
+	try {
+		const res = await fetch(`/tasks/delete/${itemId}`, {
+			method: "DELETE",
+			headers: {
+				"Content-type": "application/json",
+			},
+		});
+		if (res.ok) {
+			return res.json();
+		}
+		throw new Error("No items were deleted Something went wrong");
+	} catch (err) {
+		return err;
```

Comment:
its problematic to return different types of data from the same function.
if you return res.json and your'e code expects to handle it returning an err which is a different type of object and data can cause a complicated, hard to understand / maintain code and future bugs